timings/state.go

```diff
@@ -65,19 +65,29 @@ var timeDuration = func(start, end time.Time) time.Duration {
 // flatten flattens nested measurements into a single list within rootTimingJson.NestedTimings
 // and calculates total duration.
 func (t *Timings) flatten() interface{} {
-	if len(t.timings) == 0 {
+	var hasChangeID, hasTaskID bool
+	if t.tags != nil {
+		_, hasChangeID = t.tags["change-id"]
+		_, hasTaskID = t.tags["task-id"]
+	}
+	// store the root timing object even if it has no nested timings as long as it has a "change-id" tag
+	// but not a "task-id" tag - this signifies an "ensure" timing which we always want to record as
+	// long as it created a change.
+	if len(t.timings) == 0 && (hasTaskID || !hasChangeID) {
```

Comment:
Maybe it'd be a bit easier to follow the comment if the code read:
```
// ensure timings which created a change, have the corresponding
// change-id tag, but no task-id
isEnsureWithChange := hasChangeID && !hasTaskID

// store the root timing object ...
if len(t.timings) == 0 && !isEnsureWithChange {
    return nil
}
```

Reply:
Addressed in #6920